app/forms/import_duplicate_form.rb

```diff
@@ -29,7 +29,7 @@ def save
 
   def can_keep_both?
     # Vaccination records have lots of relationships that make it difficult to handle
-    object.is_a?(Patient)
+    object.is_a?(Patient) && !object.changesets.last&.matched_on_nhs_number
```

Comment:
```suggestion
    object.is_a?(Patient) && !object.changesets.any?(&:matched_on_nhs_number?)
```

I'm wondering if this works instead? Then it avoids the potential issue where a user resolves a duplication in the time between a changeset is created and the PDS check completes?